extensions/wikia/BannerNotifications/js/BannerNotifications.js

```diff
@@ -0,0 +1,309 @@
+		}
+		this.onCloseHandler = Function.prototype;
+	}
+
+	/**
+	 * Creates DOM element and reveals the notification
+	 * @returns {BannerNotification}
+	 */
+	BannerNotification.prototype.show = function () {
+		if (!this.hidden) {
+			return this;
+		}
+		if (!this.$element) {
+			this.$element = createMarkup(this.content, this.type);
+		}
+		this.setType(this.type);
+
+		setUpClose(this);
+
+		// Modal notifications have no close button so set a timeout
+		if (isModalShown() && typeof this.timeout !== 'number') {
+			this.timeout = defaultTimeout;
+		}
+
+		addToDOM(this.$element, this.$parent);
+
+		this.hidden = false;
+
+		// Close notification after specified amount of time
+		if (typeof this.timeout === 'number') {
+			setTimeout(function () {
+				this.hide();
+			}.bind(this), this.timeout);
+		}
+		return this;
+	};
+
+	/**
+	 * Hides a notification and removes an instance of a DOM element
+	 * @returns {BannerNotification}
+	 */
+	BannerNotification.prototype.hide = function () {
+		if (!this.hidden) {
+			fadeOut(this.$element, removeFromDOM);
+			this.$element = null;
+			this.hidden = true;
+		}
+		return this;
+	};
+
+	/**
+	 * Overrides the type of the notification
+	 * @param {String} type
+	 * @returns {BannerNotification}
+	 */
+	BannerNotification.prototype.setType = function (type) {
+		if (type !== this.type && types.hasOwnProperty(type)) {
+			this.type = type;
+			if (this.$element) {
+				this.$element
+					.removeClass(classes)
+					.addClass(type);
+			}
+		}
+		return this;
+	};
+
+	/**
+	 * Overrides the content of the notification
+	 * @param {String} content
+	 * @returns {BannerNotification}
+	 */
+	BannerNotification.prototype.setContent = function (content) {
+		if (content && content !== this.content) {
+			this.content = content;
+			if (this.$element) {
+				this.$element
+					.find('.msg')
+					.html(content);
+			}
+		}
+		return this;
+	};
+
+	/**
+	 * Generic method for indicating problem with AJAX connection
+	 * @returns {BannerNotification}
+	 */
+	BannerNotification.prototype.showConnectionError = function () {
+		return this
+			.setType('error')
+			.setContent(
+				window.mw.message('bannernotifications-general-ajax-failure')
+					.escaped()
+			)
+			.show();
+	};
+
+	/**
+	 * Allows to attach a handler to a notification close event
+	 * @param {Function} callback
+	 * @returns {BannerNotification}
+	 */
+	BannerNotification.prototype.onClose = function (callback) {
+		if (typeof callback === 'function') {
+			this.onCloseHandler = callback;
+		}
+		return this;
+	};
+
+	/**
+	 * Called once to instantiate this feature
+	 */
+	function init() {
+		if (window.skin === 'monobook') {
+			$pageContainer = $('#content');
+		} else {
+			$pageContainer = $('.WikiaPageContentWrapper');
+			headerHeight = $('#globalNavigation').height();
+			onScroll.bind(handleScrolling);
+		}
+		createBackendNotification();
+	}
+
+	/**
+	 * Creates an instance of a notification set on the backend side
```

Comment:
This message seems unclear to someone that doesn't know about notifications originating from the server. Maybe update the wording to something like: 

"Create instance of BannerNotification based on HTML passed from the server on page load."